go/types/sequence.go

```diff
@@ -16,4 +16,5 @@ type sequence interface {
 	Chunks() []Ref
 	Type() *Type
 	getCompareFn(other sequence) compareFn
+	cumulativeNumberOfLeaves(idx int) uint64 // returns the total number of leaf values reachable from this sequence for all sub-trees from 0 to |idx|
```

Comment:
Another thought... maybe we should have a `leafSequence` struct? You could save a bunch of boilerplate:
```go
type leafSequence struct {
  length int
  t *Type
  vr ValueReader
}

// implement seqLen, numLeaves, valueReader, Type, cumNumLeaves
```